overlord/devicestate/handlers_install.go

```diff
@@ -126,6 +126,40 @@ func writeLogs(rootdir string) error {
 	if err := gz.Flush(); err != nil {
 		return fmt.Errorf("cannot flush compressed log output: %v", err)
 	}
+
+	return nil
+}
+
+func writeTimings(rootdir string, st *state.State) error {
+	logPath := filepath.Join(rootdir, "var/log/install-timings.txt.gz")
+	if err := os.MkdirAll(filepath.Dir(logPath), 0755); err != nil {
+		return err
+	}
+
+	f, err := os.Create(logPath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	gz := gzip.NewWriter(f)
+	defer gz.Close()
+
+	// collect "timings" for tasks of given change
+	for _, chg := range st.Changes() {
+		// XXX: ugly, ugly, but using the internal timings requires
+		//      some refactor as a lot of the required bits are not
+		//      exported right now
```

Comment:
if chg.Kind() is "seed" ,  we want to call "debug timings"  with "--ensure=seed" not just the chg id, the latter provide timings also from the Ensure part of that which for seed is substantial